addons/website_sale/models/res_partner.py

```diff
@@ -18,8 +18,31 @@ def _compute_last_website_so_id(self):
             if website and not is_public:
                 partner.last_website_so_id = SaleOrder.search([
                     ('partner_id', '=', partner.id),
+                    ('pricelist_id', '=', partner.property_product_pricelist.id),
                     ('website_id', '=', website.id),
                     ('state', '=', 'draft'),
                 ], order='write_date desc', limit=1)
             else:
                 partner.last_website_so_id = SaleOrder  # Not in a website context or public User
+
+    @api.onchange('property_product_pricelist')
+    def _onchange_property_product_pricelist(self):
+        self.ensure_one()
+
+        open_order_count = self.env['sale.order'].search([
+            ('partner_id', '=', self._origin.id),
+            ('pricelist_id', '=', self._origin.property_product_pricelist.id),
```

Comment:
Is this true?
Eg, customer has an open SO with PL1
You change his partner PL to PL2 -> you got the warning
You change again his partner PL to PL3 -> you got no warning, but it still didn't work as expected and the warning should've been shown?

Reply:
This is how I interpreted the spec, yes. The spec explicitly asks to limit the warning to open carts. If the partner has PL2 and SO has PL1, this SO is no longer considered a cart (as per the spec).